File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        numset = set(nums)
        longest = 0;
        for num in  numset:
            if (num-1) not in numset:
                lenth = 1
                while (num+lenth) in numset:
                    lenth+=1
                longest = max(lenth,longest)
        return longest

File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(std::vector<int>& nums) {
        std::map<int,int> hash;
        for (const auto & num: nums) {
            hash[num]++;
            if (hash[num] > 1) {
                return true;
            }
        }
        return false;
    }
};